platform/pddf/platform-api-pddf-base/sonic_platform_pddf_base/pddf_sfp.py

```diff
@@ -795,249 +159,44 @@ def get_tx_disable(self):
-                tx_disable_list = []
-
-                sfpd_obj = sff8436Dom()
-                if sfpd_obj is None:
-                    return False
-
-                dom_control_raw = self.__read_eeprom_specific_bytes(
-                    QSFP_CONTROL_OFFSET, QSFP_CONTROL_WIDTH) if self.get_presence() else None
-                if dom_control_raw is not None:
-                    dom_control_data = sfpd_obj.parse_control_bytes(dom_control_raw, 0)
-                    tx_disable_list.append(
-                        'On' == dom_control_data['data']['TX1Disable']['value'])
-                    tx_disable_list.append(
-                        'On' == dom_control_data['data']['TX2Disable']['value'])
-                    tx_disable_list.append(
-                        'On' == dom_control_data['data']['TX3Disable']['value'])
-                    tx_disable_list.append(
-                        'On' == dom_control_data['data']['TX4Disable']['value'])
-
-                return tx_disable_list
-            else:
-                status_control_raw = self.__read_eeprom_specific_bytes(
-                    SFP_STATUS_CONTROL_OFFSET, SFP_STATUS_CONTROL_WIDTH)
-                if status_control_raw:
-                    data = int(status_control_raw[0], 16)
-                    tx_disable_hard = (sffbase().test_bit(
-                        data, SFP_TX_DISABLE_HARD_BIT) != 0)
-                    tx_disable_soft = (sffbase().test_bit(
-                        data, SFP_TX_DISABLE_SOFT_BIT) != 0)
-                    tx_disable = tx_disable_hard | tx_disable_soft
-
-                return tx_disable
-        else:
+        if output:
             status = int(output['status'].rstrip())
 
             if status == 1:
                 tx_disable = True
             else:
                 tx_disable = False
+        else:
+            # Use common SfpOptoeBase implementation for get_tx_disable
+            tx_disable = super().get_tx_disable()
 
         return tx_disable
 
-    def get_tx_disable_channel(self):
-        """
-        Retrieves the TX disabled channels in this SFP
-        Returns:
-            A hex of 4 bits (bit 0 to bit 3 as channel 0 to channel 3) to represent
-            TX channels which have been disabled in this SFP.
-            As an example, a returned value of 0x5 indicates that channel 0
-            and channel 2 have been disabled.
-        """
-        if not self.get_presence():
-            return 0
-
-        if self.is_osfp_port:
-            return 0
-        elif self.is_qsfp_port:
-            tx_disable_list = self.get_tx_disable()
-            if tx_disable_list is None:
-                return 0
-            tx_disabled = 0
-            for i in range(len(tx_disable_list)):
-                if tx_disable_list[i]:
-                    tx_disabled |= 1 << i
-            return tx_disabled
-        else:
-            # SFP doesnt support this
-            return 0
-
     def get_lpmode(self):
```

Comment:
why override sfpopote's [get_lpmode](https://github.com/Azure/sonic-platform-common/blob/master/sonic_platform_base/sonic_xcvr/sfp_optoe_base.py#L147). ?

Reply:
if CPLD setting for lpmode is present for any platform, that needs to be used. Else the common get_lpmode implementation of SfpOptoeBase